packages/default-reporter/test/reportingLifecycleScripts.ts

```diff
@@ -258,6 +259,124 @@ ${chalk.blue('packages/qar')} ${INSTALL}: Done`)
+      outputMaxWidth: 79,
+    },
+    streamParser: createStreamParser(),
+  })
+
+  lifecycleLogger.debug({
+    depPath: 'packages/foo',
+    optional: false,
+    script: 'node foo',
+    stage: 'preinstall',
+    wd: 'packages/foo',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/foo',
+    line: 'foo 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30',
+    stage: 'preinstall',
+    stdio: 'stdout',
+    wd: 'packages/foo',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/foo',
+    exitCode: 1,
+    optional: true,
+    stage: 'preinstall',
+    wd: 'packages/foo',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/foo',
+    optional: false,
+    script: 'node foo',
+    stage: 'postinstall',
+    wd: 'packages/foo',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/foo',
+    line: 'foo I',
+    stage: 'postinstall',
+    stdio: 'stdout',
+    wd: 'packages/foo',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/bar',
+    optional: false,
+    script: 'node bar',
+    stage: 'postinstall',
+    wd: 'packages/bar',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/bar',
+    line: 'bar I',
+    stage: 'postinstall',
+    stdio: 'stdout',
+    wd: 'packages/bar',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/foo',
+    line: 'foo II',
+    stage: 'postinstall',
+    stdio: 'stdout',
+    wd: 'packages/foo',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/foo',
+    line: 'foo III',
+    stage: 'postinstall',
+    stdio: 'stdout',
+    wd: 'packages/foo',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/qar',
+    optional: false,
+    script: 'node qar',
+    stage: 'install',
+    wd: 'packages/qar',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/qar',
+    exitCode: 0,
+    optional: false,
+    stage: 'install',
+    wd: 'packages/qar',
+  })
+  lifecycleLogger.debug({
+    depPath: 'packages/bar',
+    exitCode: 0,
+    optional: false,
+    stage: 'postinstall',
+    wd: 'packages/bar',
+  })
+
+  await expect(
+    firstValueFrom(
+      output$.pipe(map<string, string>(normalizeNewline), take(8), toArray())
+    )
+  ).resolves.toEqual([
+    `${chalk.cyan('packages/foo')} ${PREINSTALL}$ node foo`,
+    `${chalk.cyan(
+      'packages/foo'
+    )} ${PREINSTALL}: foo 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30`,
+    `${chalk.cyan('packages/foo')} ${PREINSTALL}: Failed`,
+    `${chalk.magenta('packages/qar')} ${INSTALL}$ node qar`,
+    `${chalk.magenta('packages/qar')} ${INSTALL}: Done`,
+    // `${chalk.cyan('packages/foo')} ${POSTINSTALL}$ node foo`,
+    // `${chalk.cyan('packages/foo')} ${POSTINSTALL}: foo I`,
+    `${chalk.blue('packages/bar')} ${POSTINSTALL}$ node bar`,
+    `${chalk.blue('packages/bar')} ${POSTINSTALL}: bar I`,
+    // `${chalk.cyan('packages/foo')} ${POSTINSTALL}: foo II`,
+    // `${chalk.cyan('packages/foo')} ${POSTINSTALL}: foo III`,
```

Comment:
the commented lines should be removed I guess

Reply:
sure, they were illustrating my concern [here](https://github.com/pnpm/pnpm/pull/4109#discussion_r767185290) which leads to [this](https://github.com/pnpm/pnpm/pull/4109#pullrequestreview-829557351) behavior